Mathlib/Algebra/Module/LocalizedModule.lean

```diff
@@ -1079,6 +1081,76 @@ theorem mkOfAlgebra {R S S' : Type _} [CommRing R] [CommRing S] [CommRing S'] [A
 
 end Algebra
 
+variable { A } [CommRing A] [Algebra R A] [Module A M'] [IsScalarTower R A M'] [IsLocalization S A]
+
+
+/-- If `(f : M →ₗ[R] M')` is a localization of modules, then the map
+`(localization S) × M → N, (s, m) ↦ s • f m` is the tensor product (insomuch as it is the universal
+bilinear map).
+In particular, there is an isomorphism between `LocalizedModule S M` and `(Localization S) ⊗[R] M`
+given by `m/s ↦ (1/s) ⊗ₜ m`.
+--/
+theorem isBaseChange : IsBaseChange A f := by
+  apply IsBaseChange.of_lift_unique
+  intros Q _ _ _ _ g
+  have := IsLocalizedModule.is_universal S f g <| by
+      intro s
+      rw [Module.End_isUnit_iff, Function.bijective_iff_existsUnique]
+      intro q
+      use (IsLocalization.mk' _ 1 s : A) • q
+      dsimp only
+      constructor
+      on_goal 1 => rw [Module.algebraMap_end_apply]
+      on_goal 2 =>
+        intro q' h
+        rw [Module.algebraMap_end_apply] at h
+        rw [← h, smul_comm]
+      all_goals
+        rw [← smul_assoc, Algebra.smul_def, mul_comm, IsLocalization.mk'_spec, map_one, one_smul]
```

Comment:
If you add the following lemma:
```lean
@[simp] lemma IsLocalization.smul_mk'_self {s : S} {r : R} :
    (s : R) • IsLocalization.mk' A r s = algebraMap R A r := by
  rw [IsLocalization.smul_mk', IsLocalization.mk'_mul_cancel_left]
```
(e.g., just below the your new lemma `IsLocalization.smul_mk'`) then we can golf this part to:
```suggestion
    intro s
    simp_rw [Module.End_isUnit_iff, Function.bijective_iff_existsUnique,
      Module.algebraMap_end_apply]
    intro q
    refine' ⟨(IsLocalization.mk' _ 1 s : A) • q, _, _⟩
    · simp only [← smul_assoc, IsLocalization.smul_mk'_self, map_one, one_smul]
    · rintro q rfl
      simp only [smul_comm _ (s : R), ← smul_assoc, IsLocalization.smul_mk'_self, map_one, one_smul]
```